bin/diagnostics.bash

```diff
@@ -308,15 +325,93 @@ get_config_files() {
+    TOKEN=$(fetch_oidc_token)
+    DOMAIN="https://kube.$(yq4 '.global.opsDomain' "${config["config_file_sc"]}"):6443"
+    ENDPOINT="${DOMAIN}/api/v1/namespaces/thanos/services/thanos-query-query-frontend:9090/proxy/api/v1"
+    HEADER="Authorization: Bearer ${TOKEN}"
+    RANGE_ARG="--data-urlencode start=$(date -d -"${1}" -Iseconds) --data-urlencode end=$(date -Iseconds) --data-urlencode step=1m"
+
+    query_and_parse() {
+        query="${1}"
+        print_func="${2}"
+        # shellcheck disable=SC2086
+        res=$(curl "${ENDPOINT}/query_range" -k -s --header "${HEADER}" --data-urlencode query="${query}" $RANGE_ARG)
+        if [[ $(echo "${res}" | jq 'select(.data.result != [])') ]]; then
+            for row in $(echo "${res}" | jq -c '.data.result[]'); do
+                "${print_func}" "${row}" "${@:3}"
+            done
+        fi
+    }
+
+    # shellcheck disable=SC2317
+    print_dates() {
+        eval "${2}"
+        echo "${1}" | jq '.values[][0]' | xargs -I {} date -d@{}
+        echo
+    }
+
+    # Fluentd output error/retry rate
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+    echo "Querying fluentd output error rate."
+    query_and_parse 'sum(rate(fluentd_output_status_retry_count[1m])) > 0' print_dates 'echo "Fluentd output error rate over 0 on the dates:"'
+
+
+    # Dropped packets going from pod
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+    echo "Querying dropped packages."
+    # shellcheck disable=SC2016
+    query_and_parse 'rate(no_policy_drop_counter[1m]) > 0' print_dates 'echo "Found dropped packages going $([[ $(echo "${row}" | jq -r .metric.type) == "fw" ]] && echo "from" || echo "to") pod: $(echo "${row}" | jq '.metric.exported_pod') on dates:"'
+
+
+    # Uptime
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+    echo "Querying uptime."
+    # shellcheck disable=SC2016
+    query_and_parse 'max by (target) (probe_success) < 1' print_dates 'echo "The target $(echo "${row}" | jq '.metric.target') was down on dates:"'
+
+
+    # Opensearch status <instant Query>
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+    echo "Querying Opensearch cluster status"
+    # shellcheck disable=SC2086
+    res=$(curl "${ENDPOINT}/query" -k -s --header "${HEADER}" --data-urlencode query='elasticsearch_cluster_health_status{color=~"yellow|red"} > 0')
+    if [[ $(echo "${res}" | jq 'select(.data.result != [])') ]]; then
+        echo "Opensearch is in $(echo "${res}" | jq '.data.result[0].metric.color') state!"
+    fi
+}
+
+# run_diagnostics_query_metric <metric>
+run_diagnostics_query_metric() {
+    TOKEN=$(fetch_oidc_token)
+    DOMAIN="https://kube.$(yq4 '.global.opsDomain' "${config["config_file_sc"]}"):6443"
+    ENDPOINT="${DOMAIN}/api/v1/namespaces/thanos/services/thanos-query-query-frontend:9090/proxy/api/v1"
+    HEADER="Authorization: Bearer ${TOKEN}"
+
+    curl "${ENDPOINT}/query" -k --header "${HEADER}" --data-urlencode query="${1}" | jq
```

Comment:
non-blocker: Consider creating a "query thanos using curl" wrapper to avoid repeating building it.